lib/commands/exam.js

```diff
@@ -206,50 +217,75 @@ module.exports = TestCommand.extend({
    */
   _generateCustomConfigs(commandOptions) {
     const config = this._super._generateCustomConfigs.apply(this, arguments);
+    const events = config.custom_browser_socket_events || {};
+
+    this._addBrowserSocketEvents(config, events);
+
+    if (commandOptions.loadBalance || commandOptions.replayExecution) {
+      this._addLoadBalancingBrowserSocketEvents(events, this.commands, this.testemEvents);
+    }
+
+    config.custom_browser_socket_events = events;
 
     if (!commandOptions.loadBalance && !commandOptions.replayExecution && !commandOptions.parallel )
       return config;
 
     config.testPage = getMultipleTestPages(config, commandOptions);
 
-    if (commandOptions.loadBalance || commandOptions.replayExecution) {
-      config.custom_browser_socket_events = this._addLoadBalancingBrowserSocketEvents(
-        config
+    if (commandOptions.replayExecution) {
+      this.testemEvents.setReplayExecutionMap(
+        commandOptions.replayExecution,
+        commandOptions.replayBrowser
       );
-      if (commandOptions.replayExecution) {
-        this.testemEvents.setReplayExecutionMap(
-          commandOptions.replayExecution,
-          commandOptions.replayBrowser
-        );
-      }
     }
 
     return config;
   },
 
   /**
-   * Returns a object of event handlers which enables load balancing.
-   * These event handlers will be registered on Testem's browserTestRunner socket instance
+   * Add events to enable to send and receive module metadata
    *
    * @param {Object} config
-   * @return {Object} events
+   * @param {Object} events
    */
-  _addLoadBalancingBrowserSocketEvents(config) {
+  _addBrowserSocketEvents(config, events) {
```

Comment:
Could we rename this to `_setupAndGetBrowserSocketEvents`?
I feel this reflects the code more.